src/ComposerCompatibleUpdater.php

```diff
@@ -64,4 +70,15 @@ public function getLength(string $target): int
             throw new NotFoundException($target, 'Target');
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function getMetadataForTarget(string $target): ?TargetsMetadata
```

Comment:
This still needs tests, correct?

so this is change is so we don't have to search for the metadata more than once and the Loader change is so we don't load files more than once, correct?

Reply:
Correct on all counts.

That said, I'm thinking of removing this bit from the PR because it's very hard to test.